### ISSUE TITLE
WorkflowEmailService Error logging when Sender address not set in form workflow

### ISSUE THREAD

**Issue by nvisage-gf** [umbraco/Umbraco.Forms.Issues#439]
Umbraco Forms 8.5.4

When sending email via the `WorkflowEmailService` an exception is logged if the form workflow sender address does not contain `@`; the following line then sets the Sender address to the fallback `NotificationEmailAddress` - so now it does contain an "at".

```
39        if (!sender.Contains("@"))
40          this._logger.Error<WorkflowEmailService>("Error sending email, invalid sender address {Sender}", (object) sender);
41        message.From = new MailAddress(string.IsNullOrEmpty(sender) ? this._contentSection.NotificationEmailAddress : sender);
```

The check & logging of exception/warning should be amended - otherwise all email workflows generate an error which can be confusing when monitoring Forms for delivery issues.

We have applied 8.5.4 to two environments which both show this exception in the logs since applying; verified locally that email is sent using fallback.

Refs: https://github.com/umbraco/Umbraco.Forms.Issues/issues/406

**Comment by ronaldbarendse** [comment]
Having the `SenderEmail` configurable by editors shouldn't be possible if you ask me: https://github.com/umbraco/Umbraco.Forms.Issues/issues/139.

If you've correctly configured your SMTP account to send using a specific email address or domain (in both `configuration/system.net/mainSettings/smtp@from` and the Umbraco `settings/content/notifications/email` setting), changing this address will likely cause errors when sending the mail or otherwise issues with SPF validation, etc. Because Exchange/Office 365 will by default only allow the authenticated SMTP account/address to be used as sender, having an editor set it to anything else will break sending mails.

If you want to configure another address to reply to, it's now possible to set the `ReplyToEmail` instead: https://github.com/umbraco/Umbraco.Forms.Issues/issues/51.

**Comment by ronaldbarendse** [comment]
I can confirm this error is always logged when there's no `SenderEmail` configured in the email workflows (both basic and Razor, as they both use the `IWorflowEmailService`). This is indeed introduced by changes to fix https://github.com/umbraco/Umbraco.Forms.Issues/issues/406.

Although not exactly the same, the error message logged when an invalid reciever email is configured is also incorrect: `Error in 'reply to', invalid address "[$formEmail]"` (it's not the 'reply to' field).

**Comment by nul800sebastiaan** [comment]
Thanks, this is now corrected to first get the email from the UmbracoSettings.config if it was empty before, if it still doesn't contain an `@` this will be logged again of course, but not when the workflow doesn't have a SenderEmail configured.

**Comment by nvisage-gf** [comment]
This is still present in 8.6.0; every email sent via the default WorkflowEmailService logs a false positive exception if Sender address is not passed to method due to minor logic bug:

`address` variable should be checked not `sender` here:

```
 string address = string.IsNullOrEmpty(sender) ? this._contentSection.get_NotificationEmailAddress() : sender;
        if (!sender.Contains("@"))
          LoggerExtensions.Error<WorkflowEmailService>(this._logger, "Error sending email, invalid sender address {Sender}", new object[1]
          {
            (object) sender
          });
```

**Comment by ronaldbarendse** [comment]
The most confusing part is that the mail is sent, even after logging the error!

The check is also quite useless, as `test@` still isn't a valid address and still causes a `FormatException` to be thrown from the `MailAddress` constructor. Why not catch this exception and provide some additional context about in what type of address it was encountered (e.g. sender/from, to or reply-to). Logging the actual email addresses should probably also be avoided because of GDPR/privacy concerns.

**Comment by hfloyd** [comment]
I agree that this is a bothersome issue. I am trying to debug why emails are not getting sent on form submission on a client site, and having "red herring" error messages in the log is really unhelpful/misleading.

**Comment by ChilliHicks** [comment]
This is still happening in v8.7.2. Please re-open and fix, thanks.

**Comment by DarkSummon** [comment]
I can confirm this is still occurring in version 8.7.2.

**Comment by nikolajlauridsen** [comment]
We've taken a look at this again and implemented a fix, that'll be in the next release, 8.6.1
